### PR TITLE
Where Jón makes more work on the client string

### PR DESCRIPTION
**Pull request by jontg**

### CODE REVIEW

File: soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java [hunk 1]
```diff
@@ -8,20 +8,33 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.slf4j.MDC;
 
-/**
- * Created by ankush.goyal on 2/14/15.
- */
 @Singleton
 public class MDCContextFilter implements Filter {
+	private static final Logger log = LoggerFactory.getLogger(MDCContextFilter.class);
```
Comment (relateiq-ec2) [inline soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java]: MINOR Rename the "log" logger to comply with the format "LOG(?:GER)?". [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1312)
MAJOR Remove this unused "log" private field. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

**Comment by relateiq-ec2** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java [hunk 2]
```diff
@@ -8,20 +8,29 @@
...
-/**
- * Created by ankush.goyal on 2/14/15.
- */
 @Singleton
 public class MDCContextFilter implements Filter {
 	public static final String USER_ID = "userId";
+	public static final String CLIENT_HEADER = "RIQ-Client";
+	public static final String CLIENT = "client";
+
+	public static final ImmutableSet headerProvideableClients = ImmutableSet.of(
```
Comment (keven425) [inline soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java]: @jontg is there a way to grab these constants from polylith? I'm not too familiar with this.

File: soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java [hunk 3]
```diff
@@ -36,10 +45,18 @@ public void doFilter(ServletRequest servletRequest,
 		if (context != null) {
 			// padding so things can collapse in log line when these are null
 			MDC.put(USER_ID, " " + context.toString());
+			if (servletRequest instanceof HttpServletRequest) {
+				MDC.put(CLIENT,
+						Optional.ofNullable(((HttpServletRequest) servletRequest).getHeader(CLIENT_HEADER))
+								.filter(headerProvideableClients::contains)
+								.map(s -> " client=" + s)
```
Comment (keven425) [inline soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java]: @jontg I saw a comment on line 46: `padding so things can collapse in log line when these are null`. I assume the padding is also added for client log line?

File: soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java [hunk 4]
```diff
@@ -36,10 +45,18 @@ public void doFilter(ServletRequest servletRequest,
...
+								.orElse(" client=anon"));
+			}
 		}
 
 		filterChain.doFilter(servletRequest, servletResponse);
 
+		MDC.remove(CLIENT);
```
Comment (keven425) [inline soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java]: Why are we removing it, immediately after putting it? (again, I'm not familiar. thanks for explaining in advance)

File: soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java (hunk 2 above)
Comment (jontg) [inline soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java]: No good way.  I can move the constants into soa, but polylith is supposed
to be deprecated, so adding more dependencies to it isn't a good idea.

Comment (keven425) [inline soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java]: Sounds good.

**Comment by keven425** [comment]
@jontg this looks good to me. feel free to merge!

File: soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java (hunk 4 above)
Comment (jontg) [inline soa-auth/src/main/java/com/relateiq/soa/auth/MDCContextFilter.java]: the bulk of the request happens in the line above, in the `doFilter(...)` call; so while it looks like we're removing it right away, but it's actually happening at the end of the request.

**Pull request merged**